### PR TITLE
Add some additional lints

### PR DESCRIPTION
**Pull request by TimWhiting**
Migrated a few deprecated analyzer apis. Handle dependencies for constructor tearoffs and assignment of ref to objects using cascades.

Still not handling escape of ref via functions. Will work on that next.

### CODE REVIEW

**Comment by rrousselGit** [comment]
Nice! The CI seem to be failing though

**Comment by rrousselGit** [comment]
> I've added hot-reload support to the linter for local testing / debugging, a lot easier than using the `custom_lint.log` file & exceptions / print statements, because you can use breakpoints and the debugger. Happy to contribute that to `custom_lint`, but a little more generalized. WDYT?

Sure. Although I honestly don't quite understand what you meant

**Comment by TimWhiting** [comment]
This is the relevant code and generalization to support fast debugging 
for any custom lint plugin.

https://github.com/invertase/dart_custom_lint/pull/26#issue-1345394190

**Comment by TimWhiting** [comment]
Let's squash and merge this. I'm continuing work on it, but I want to let others contribute, and this is in a good state as it is. The build is failing for the same reason the build is failing on the master branch.

File: examples/counter/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+analyzer:
+  # plugins:
```
Comment (rrousselGit) [inline examples/counter/analysis_options.yaml:2]: Why add custom_lint to disable it?

File: packages/riverpod/example/lib/models.dart [hunk 2]
```diff
@@ -45,6 +45,9 @@ class Repository {
         'hash': hash,
       },
     );
+    if (result.data == null) {
```
Comment (rrousselGit) [inline packages/riverpod/example/lib/models.dart:48]: Why this change?

File: examples/counter/analysis_options.yaml (hunk 1 above)
Comment (rrousselGit) [reply examples/counter/analysis_options.yaml:2]: Ah wait since you're based on my branch, that could be me.
